schema.go

```diff
@@ -17,7 +17,7 @@ import (
 	jsoniter "github.com/json-iterator/go"
 )
 
-var nullDefault = struct{}{}
+var nullDefault null = struct{}{}
```

Comment:
I am not sure I agree with this change. I think it would be better to deal with `nil` type in the default encoder:
```
defaultType := reflect2.TypeOf(def)
if defaultType == nil {
	defaultType = reflect2.TypeOf((*null)(nil))
}
```

Reply:
>  I think it would be better to deal with nil type in the default encoder

This approach is much better as it reduces the scope of changes in the PR.